chat-spaceの自動更新機能

1003Hisashi opened the pull request:
#What
chat-spaceの自動更新機能追加

#Why
チャットが自動で更新されるようにしましょう。

Commit:
create pull request

1003Hisashi commented:
#What 
gyazo動画

#why
動作チェックのため

https://gyazo.com/d54a92719c7de17969c1685cde9265f5

Commit:
initial commit

TOMARUMARU commented:
gyazo見ると、ページをリロードしているように見えるのですが、非同期通信はできていますでしょうか？
ページをリロードしなくても更新できれば非同期できていることになります。

Commit:
非同期できていなかったのでコード修正

#What
コード修正

#Why
非同期できた

1003Hisashi commented:
コード修正し、非同期通信できるようになりました。
片方でメッセージ送信したとき、もう片方で日時がundefinedになり
更新すると表示されることが気になります。

日付も表示できるようにするヒントなどご教授いただければ幸いです。

[https://gyazo.com/289b00bc66d4b47369cb58cfbf11c4d5](url)

nakamotoyuto commented on app/assets/javascripts/message.js:
```diff
@@ -1,65 +1,59 @@
+              ${message.user_name}
+            </div>
+            <div class="upper-message__date">
+              ${message.date}
```
json.jbuilderファイルでは
> json.created_at @message.created_at.strftime("%Y年%m月%d日 %H時%M分")

とかかれていますので
message.created_atにしましょう！

nakamotoyuto commented on app/assets/javascripts/message.js:
```diff
@@ -1,65 +1,59 @@
+          $('.messages').append(insertHTML);
+          $('.messages').animate({scrollTop: $('.messages')[0].scrollHeight});
+          $("#new_message")[0].reset();
+          $(".form__submit").prop("disabled", false);
```
こちらは不要なので削除しましょう！

nakamotoyuto reviewed (commented):
訂正& 添付ありがとうございます！

コメントさせていただきました！
確認お願いいたします！

Commit:
日時がundefinedになる件の修正

#What
message.jsファイルの修正

#Why
日時がundefinedになる件の修正
テストし日時が表示されることを確認

1003Hisashi commented:
#What
message.jsファイルの修正

#Why
日時がundefinedになる件の修正
テストし日時が表示されることを確認

[https://gyazo.com/c1ef713a35cdca8216098ddbd4e5ffad](url)

nakamotoyuto commented on app/assets/javascripts/message.js:
```diff
@@ -1,65 +1,58 @@
+          });
+          $('.messages').append(insertHTML);
+          $('.messages').animate({scrollTop: $('.messages')[0].scrollHeight});
+          $("#new_message")[0].reset();
```
こちらは不要なので削除しましょう！

1003Hisashi replied:
3行とも削除しました

nakamotoyuto commented on app/assets/javascripts/message.js:
```diff
@@ -1,65 +1,58 @@
-  });
-});
\ No newline at end of file
+    image = (message.image) ? `<img class= "lower-message__image" src=${message.image} >` : ""; //三項演算子を使ってmessage.imageにtrueならHTML要素、faiseなら空の値を代入。
```
varをつけましょう！

1003Hisashi replied:
var image~に修正しました

nakamotoyuto reviewed (commented):
訂正コメント出させていただきました！
確認お願いいたします！

Commit:
不要な記述を削除

#What
message.jsにおいて不要な記述削除

1003Hisashi commented:
[https://gyazo.com/2b7183006704544125e6d212a09f8701](url)

miyagit commented on app/views/api/messages/index.json.jbuilder:
```diff
@@ -0,0 +1,7 @@
+json.array! @messages do |message|
+  json.content message.content
+  json.image message.image.url
+  json.created_at message.created_at.strftime("%Y年%m月%d日 %H時%M分")
```
strftimeでも大丈夫なのですがtime_formatというものが便利なので、参考程度にtime_formatの使い方も見てみましょう。
下記のサイトに使い方が載っているので参考になりそうです。
https://pikawaka.com/rails/strftime

1003Hisashi replied:
アドバイスありがとうございます！

miyagit commented on app/views/groups/_form.html.haml:
```diff
@@ -1,3 +1,59 @@
+-# = form_for group do |f|
```
こちらコメントアウトする理由を書きましょう。

1003Hisashi replied:
コードを書き直す前のものをコメントアウトして、
書き直した後のコードで問題ないか確認したかったので一旦コメントアウトしていました。
必要ないと判断したので、コメントアウトした部分を削除しています。

miyagit replied:
承知しました。それではこちらのコメントアウトまだ削除されていないので削除お願いします。

1003Hisashi replied:
削除完了しました

miyagit replied:
こちらコメントアウト削除されていないのですが、プッシュのしわすれでしょうか？

1003Hisashi replied:
プッシュ時にエラーが出ていたのを解決し忘れていました。
再度プッシュしましたが、いかがでしょうか？

[https://github.com/1003Hisashi/chat-space/pull/11/commits/8f239123131f6fc2518aedb7131786b73cfa92db](url)

miyagit commented on app/views/messages/create.json.jbuilder:
```diff
@@ -1,5 +1,5 @@
 json.content @message.content
+json.image @message.image_url
 json.created_at @message.created_at.strftime("%Y年%m月%d日 %H時%M分")
```
strftimeでも大丈夫なのですがtime_formatというものが便利なので、参考程度にtime_formatの使い方も見てみましょう。
下記のサイトに使い方が載っているので参考になりそうです。
https://pikawaka.com/rails/strftime

1003Hisashi replied:
確認しました。今回は2箇所だけなのでそのままにしますが、時刻表示する箇所がもっと増えれば活用します。

Commit:
部分修正

#What
以下のファイルの記述を部分修正
message.js
index.json.jbuilder
index.htmal.haml
routes.rb

#Why
①自動更新ができなくなったため、できるようになる状態まで修正
②余分なコメントアウトの削除

1003Hisashi commented:
自動更新ができていることを確認した動画です。

[https://gyazo.com/9f8694c5dd70e396b5fe426277367189](url)

Commit:
不要なコメントアウト削除

#What
不要なコメントアウト削除

ken-nag commented:
lgtm
